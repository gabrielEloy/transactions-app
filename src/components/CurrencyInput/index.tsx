import React from 'react';
import {TextInputProps} from 'react-native';

import {TextInputMask} from 'react-native-masked-text';

const CurrencyInput = (props: TextInputProps) => {
  return (
    <TextInputMask
      type="money"
      options={{
        precision: 2,
        separator: ',',
        delimiter: '.',
        unit: 'R$ ',
        suffixUnit: '',
      }}
      {...props}
    />
  );
};

export default CurrencyInput;
