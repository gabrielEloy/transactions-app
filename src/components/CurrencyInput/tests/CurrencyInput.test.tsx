import React from 'react';
import {render} from 'react-native-testing-library';
import CurrencyInput from '../index';

describe('Button', () => {
  it('should render correctly', () => {
    const tree = render(<CurrencyInput />);
    expect(tree).toMatchSnapshot();
  });
});
