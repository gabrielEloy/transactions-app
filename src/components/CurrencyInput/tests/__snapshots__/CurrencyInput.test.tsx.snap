// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render correctly 1`] = `
<TextInput
  allowFontScaling={true}
  keyboardType="numeric"
  onChangeText={[Function]}
  options={
    Object {
      "delimiter": ".",
      "precision": 2,
      "separator": ",",
      "suffixUnit": "",
      "unit": "R$ ",
    }
  }
  rejectResponderTermination={true}
  type="money"
  underlineColorAndroid="transparent"
  value=""
/>
`;
